database/pgSeed.js

```diff
@@ -0,0 +1,102 @@
+// seed 10M rows
+const faker = require('faker');
+const fs = require('file-system');
+const _progress = require('cli-progress');
+
+const randomLorem = faker.lorem.word;
+const productType = [
+  'Chair',
+  'Dresser',
+  'Drawer',
+  'Desk',
+  'Table',
+  'Island',
+  'Rug',
+  'Towel',
+  'Lamp',
+  'Box',
+  'Storage',
+  'Mirror',
+  'Stool',
+  'Bench',
+  'Curtain',
+  'Organizer',
+  'Bowl',
+  'Cutlery'
+];
+const dimensions = [
+  'null',
+  '41 3/8x19 5/8',
+  '28 3/4x19 5/8',
+  '51 5/8x23 5/8',
+  '47 1/4',
+  '32 5/8x37x35 3/8',
+  '18 1/8x13 3/4',
+  '15 3/8x16 1/8',
+  '23 5/8x15 3/8',
+  '21 1/4x26'
+];
+
+const createProduct = function(id) {
+  var result = '';
+  var randomProduct =
+    productType[Math.floor(Math.random() * productType.length)];
+  var randomDim = dimensions[Math.floor(Math.random() * dimensions.length)];
+  result +=
+    id.toString() +
+    ',' +
+    randomLorem().toUpperCase() +
+    ',' +
+    randomProduct +
+    ',' +
+    randomDim;
+  return result;
+};
+
+const count = 10000000;
+const file = 'database/ikea_seed.csv';
+
+// cli-progress bar
+const bar = new _progress.Bar({}, _progress.Presets.shades_grey);
+
+// if file already exists
+if (fs.existsSync(file)) {
+  fs.unlinkSync(file);
+}
+
+bar.start(count, 0);
+const stream = fs.createWriteStream(file);
+stream.on('err', err => console.log(err));
+stream.on('close', () => {
+  bar.stop();
+  console.timeEnd();
+  console.log('Successfully wrote file!');
+});
+
+write10Mil();
+function write10Mil() {
+  console.time();
+  let i = count;
+  // initiate the csv file with headers
+  writeFile();
+  // stream.write('id,name,type,dimensions\n');
+  function writeFile() {
+    let ok = true;
+    do {
+      i--;
+      // write the very last entry
+      if (i === 0) {
+        stream.write(createProduct(10000000 - i));
+        stream.end();
+      } else {
+        ok = stream.write(createProduct(10000000 - i) + '\n');
```

Comment:
More curiosity: What is the intention when reseting 'ok' here?

Reply:
i got this format straight from the stream.write example on node and if i understand it correctly, it's just constantly checking if stream.write succeeded.
so if it fails and ok evaluates to false, it'll break out of the while loop and go into the if statement on line 98

Reply:
Thanks for sharing.